netlogo-gui/src/main/lab/gui/ProgressDialog.scala

```diff
@@ -139,6 +136,44 @@ private [gui] class ProgressDialog(dialog: java.awt.Dialog, supervisor: Supervis
       workspace.jobManager.finishSecondaryJobs(null)
     }
   }
+
+  def updateView(check:Boolean): Unit = {
+    displaySwitch.setSelected(check)
+    workspace.displaySwitchOn(check)
+  }
+
+  def setUpdateView(status:Boolean): Unit = {
+    if(status){
+      updateView(true)
+    }
+    else {
+      updateView(false)
+    }
+  }
+
+  def plotsAndMonitorsSwitch(check:Boolean): Unit = {
+    if(check){
+      plotsAndMonitorsSwitch.setSelected(true)
+      workspace.setPeriodicUpdatesEnabled(true)
+    }
+    else {
+      plotsAndMonitorsSwitch.setSelected(false)
+      workspace.setPeriodicUpdatesEnabled(false)
+      workspace.jobManager.finishSecondaryJobs(null)
+    }
+  }
+
+  def setPlotsAndMonitorsSwitch(status:Boolean): Unit = {
+    if(status){
+      plotsAndMonitorsSwitch(true)
+      updatePlots = true
+    }
+    else {
+      plotsAndMonitorsSwitch(false)
+      updatePlots = false
+    }
```

Comment:
Method was supposed to work in this way,make proper changes and produce the expected results or there is a better way of doing this ?

Reply:
This method can be rewritten to be two lines instead of 6, without changing how it functions. When you say `plotsAndMonitorsSwitch(false)` and `updatePlots = false`, what is the value of `status`?

Reply:
Got your point!
Instead using if-else , we can directly pass status variable. 
Will make changes and push it soon. 
Thanks for your suggestion.